src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -201,6 +230,23 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
                                       completion:(MTRDeviceOpenCommissioningWindowHandler)completion
     MTR_AVAILABLE(ios(17.0), macos(14.0), watchos(10.0), tvos(17.0));
 
+/**
+ * Requests diagnostic logs of the desired types from the device.
+ *
+ * @param types      The types of logs being requested. This should correspond to values in the enum MTRDiagnosticLogType.
+ * @param timeout    The timeout in minutes for getting the logs.  If the timeout expires, completion
```

Comment:
Should make it clear whether fractional minutes are allowed.  But also: why not just have this be NSNumber holding NSTimeInterval, document that, and then we don't need to do the documenting of units...

Though I do wonder whether passing `NSTimeInterval *` would work, and how that maps to Swift.

Reply:
changed it to NSTimeInterval and added documentation if set to 0 there will be no timeout